### PR TITLE
feat(gaussdb/opengauss): support new ha mode for centralized type

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
1. support new ha mode for centralized type and related parameter: replica_num.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support new ha mode for centralized type.
2. update the related document and acc test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccOpenGaussInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccOpenGaussInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_basic
=== PAUSE TestAccOpenGaussInstance_basic
=== CONT  TestAccOpenGaussInstance_basic
--- PASS: TestAccOpenGaussInstance_basic (1740.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1740.517s
```
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccOpenGaussInstance_centralizedHAMode'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccOpenGaussInstance_centralizedHAMode -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_centralizedHAMode
=== PAUSE TestAccOpenGaussInstance_centralizedHAMode
=== CONT  TestAccOpenGaussInstance_centralizedHAMode
--- PASS: TestAccOpenGaussInstance_centralizedHAMode (1516.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1516.391s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_haModeCentralized'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_haModeCentralized -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_haModeCentralized
=== PAUSE TestAccOpenGaussInstance_haModeCentralized
=== CONT  TestAccOpenGaussInstance_haModeCentralized
--- PASS: TestAccOpenGaussInstance_haModeCentralized (1768.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1768.195s
```

File: docs/resources/gaussdb_opengauss_instance.md [hunk 1]
```diff
@@ -8,19 +8,26 @@ GaussDB OpenGauss instance management within HuaweiCoud.
...
-  password          = "Test@123"
+variable "network_id" {}
+variable "security_group_id" {}
+variable "instance_name" {}
+variable "instance_password" {}
+
+data "huaweicloud_availability_zones" "test" {}
+
+resource "huaweicloud_gaussdb_opengauss_instance" "test" {
+  network_id        = var.network_id
```
Comment (niuzhenguo) [inline docs/resources/gaussdb_opengauss_instance.md]: as we discussed, keep `subnet_id` for now.

File: docs/resources/gaussdb_opengauss_instance.md [hunk 2]
```diff
@@ -35,137 +44,218 @@ resource "huaweicloud_gaussdb_opengauss_instance" "instance_acc" {
...
-  reference to the API docs for valid options.
+* `backup_strategy` - (Optional, List) Specifies the advanced backup policy.
+  The [object](#opengauss_backup_strategy) structure is documented below.
 
-The `volume` block supports:
+<a name="opengauss_ha"></a>
+The `ha` block supports:
 
-* `type` - (Required, String, ForceNew) Specifies the volume type. Only "ULTRAHIGH" is supported now.
+* `mode` - (Required, String, ForceNew) Specifies the database mode. Only **enterprise** is supported now.
```
Comment (niuzhenguo) [inline docs/resources/gaussdb_opengauss_instance.md]: get rid of `Only enterprise is supported now`

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 3]
```diff
@@ -49,8 +64,9 @@ func ResourceOpenGaussInstance() *schema.Resource {
 				Computed: true,
 			},
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: utils.SuppressCaseDiffs,
```
Comment (niuzhenguo) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: why supress case diff here for name?

**Pull request merged**